src/Symfony/Component/Security/Core/User/UserProviderInterface.php

```diff
@@ -30,6 +30,8 @@
  * @see UserInterface
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @method bool upgradePassword(UserInterface $user, string $newEncodedPassword)
```

Comment:
I don't think we should make it part of the UserProviderInterface. There are lots of valid use cases for *not* dealing with a password in your User (see the discussions from the EU-FOSSA hackathon about that). It should be a separate optional interface implemented by the provider IMO.

Reply:
This was also my first impression. Next easiest thing to try is to make it an optional interface that a user provider *may* implement - an extra super-power. I find user providers confusing in general, but the "user provider system" indeed seems like the proper place for this at this time.

Reply:
fixed by moving the method to a new `PasswordUpgraderInterface`